src/In.ProjectEKA.TMHHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,534 @@
+                "yyyy-MM-ddTHH:mm:ss.ffff", "yyyy-MM-ddTHH:mm:ss", "yyyy-MM-ddTHH:mm:ss.fffff",
+                "yyyy-MM-ddTHH:mm:ss.ffffff", "dd/MM/yyyy", "dd/MM/yyyy hh:mm:ss", "dd/MM/yyyy hh:mm:ss tt",
+                "dd/MM/yyyyTHH:mm:ss.fffzzz"
+            };
+            var tryParseExact = DateTime.TryParseExact(dateString,
+                formatStrings,
+                CultureInfo.CurrentCulture,
+                DateTimeStyles.None,
+                out var aDateTime);
+            if (!tryParseExact)
+            {
+                Log.Error($"Error parsing date: {dateString}");
+            }
+
+            return aDateTime;
+        }
+
+
+        private static async Task<Option<MedicationResponse>> FindMedicationRequestData(DataRequest dataRequest,
+            List<Prescription> prescriptions, string patientName)
+        {
+            LogDataRequest(dataRequest);
+            var list = new List<IMedication>();
+
+            foreach (var prescription in prescriptions)
+            {
+                if (!WithinRange(dataRequest.DateRange, prescription.Date))
+                {
+                    continue;
+                }
+
+                var uuid = Uuid.Generate().Value;
+                var id = uuid.Split(":").Last();
+                var uuidMedication = Uuid.Generate().Value;
+                var medicationId = uuid.Split(":").Last();
+                var medicationRequestRepresentation = new MedicationRequestRepresentation
+                {
+                    FullUrl = uuid,
+                    Resource = new MedicationRequestResource
+                    {
+                        Id = id,
+                        Intent = "order",
+                        Status = "active",
+                        Subject = new Subject(patientName),
+                        AuthoredOn = prescription.Date.Date,
+                        ResourceType = HiType.MedicationRequest,
+                        DosageInstruction = new DosageInstruction(1, prescription.Dosage),
+                        MedicationReference = new MedicationReference("Medication/" + medicationId, "Medications")
+                    }
+                };
+                var coding =
+                    new Coding(prescription.GenName, prescription.Medicine);
+                var medicationRepresentation = new MedicationRepresentation
+                {
+                    FullUrl = uuidMedication,
+                    Resource = new MedicationResource
+                    {
+                        Id = medicationId, ResourceType = HiType.Medication,
+                        Code = new Code(new List<Coding> {coding}, prescription.Medicine)
+                    }
+                };
+
+                list.Add(medicationRequestRepresentation);
+                list.Add(medicationRepresentation);
+            }
+
+            if (!list.Any()) return Option.None<MedicationResponse>();
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4940156)